internal/stats/stats_service.go

```diff
@@ -460,6 +512,46 @@ func makePercentMD(prefix, metric string) metricDescriptor {
 	}
 }
 
+func makeNearCacheMSMD(ncName, metric string) metricDescriptor {
+	return metricDescriptor{
+		Prefix:             nearCacheDescriptorPrefix,
+		Metric:             metric,
+		Discriminator:      nearCacheDescriptorDiscriminator,
+		DiscriminatorValue: ncName,
+		HasUnit:            true,
+		Unit:               metricUnitMS,
+	}
+}
+
+func makeNearCacheCountMD(ncName, metric string) metricDescriptor {
+	return metricDescriptor{
+		Prefix:             nearCacheDescriptorPrefix,
+		Metric:             metric,
+		Discriminator:      nearCacheDescriptorDiscriminator,
+		DiscriminatorValue: ncName,
+		HasUnit:            true,
+		Unit:               metricUnitCount,
+	}
+}
+
 func makeTextStat(md *metricDescriptor, value interface{}) stat {
 	return stat{k: md.String(), v: fmt.Sprintf("%v", value)}
 }
+
+var ncMSMetrics = map[string]func(s nearcache.Stats) int64{
+	"creationTime":            func(s nearcache.Stats) int64 { return s.CreationTime.UnixMilli() },
+	"lastPersistenceTime":     func(s nearcache.Stats) int64 { return s.LastPersistenceTime.UnixMilli() },
+	"lastPersistenceDuration": func(s nearcache.Stats) int64 { return s.LastPersistenceDuration.Milliseconds() },
+}
+var ncCountMetrics = map[string]func(s nearcache.Stats) int64{
+	"evictions":                   func(s nearcache.Stats) int64 { return s.Evictions },
+	"hits":                        func(s nearcache.Stats) int64 { return s.Hits },
+	"lastPersistenceKeyCount":     func(s nearcache.Stats) int64 { return s.LastPersistenceKeyCount },
+	"lastPersistenceWrittenBytes": func(s nearcache.Stats) int64 { return s.LastPersistenceWrittenBytes },
+	"misses":                      func(s nearcache.Stats) int64 { return s.Misses },
+	"ownedEntryCount":             func(s nearcache.Stats) int64 { return s.OwnedEntryCount },
+	"expirations":                 func(s nearcache.Stats) int64 { return s.Expirations },
+	"invalidations":               func(s nearcache.Stats) int64 { return s.Invalidations },
+	"invalidationRequests":        func(s nearcache.Stats) int64 { return s.InvalidationRequests },
+	"ownedEntryMemoryCost":        func(s nearcache.Stats) int64 { return s.OwnedEntryMemoryCost },
+}
```

Comment:
lastPersistenceFailure is missing as far as I see. Will you add it later?

Reply:
That's not sent with the binary statistics.